### PR TITLE
Refactor: use variables to make lines more readable

### PR DESCRIPTION
**Pull request by till**
Introduced two variables for things that get used and used again
inside the template. Hopefully shorting lines, so we can break it
down into sub-templates later.

### CODE REVIEW

File: haproxy/templates/default/haproxy.cfg.frontend.erb [hunk 1]
```diff
@@ -1,33 +1,36 @@
...
-  <% unless @node.fetch('haproxy',{}).fetch('ssl', 'off') == 'off'
+  <% unless haproxy_cfg.fetch('ssl', 'off') == 'off'
     certificate = "#{@node['ssl-deploy']['directory']}/cert.combined.pem"
   -%>
   bind *:443 ssl crt <%= certificate %>
   reqadd X-Forwarded-Proto:\ https if { ssl_fc }
-  <% unless @node.fetch('haproxy',{}).fetch('hsts',{}).nil? -%>
-  rspadd Strict-Transport-Security:\ max-age=<%=@node.fetch('haproxy',{}).fetch('hsts',{})%> if { ssl_fc }
+  <% unless haproxy_cfg.fetch('hsts',{}).nil? -%>
+  rspadd Strict-Transport-Security:\ max-age=<%=haproxy_cfg.fetch('hsts',{})%> if { ssl_fc }
```
Comment (till) [inline haproxy/templates/default/haproxy.cfg.frontend.erb]: @fh When I read this statement, I was wondering too if this ever works.

1. Line 12, it seems to default to an object by default, but we check for nil.
2. Line 13, I would expect this to be a string/number and also not an object.

File: haproxy/templates/default/haproxy.cfg.frontend.erb [hunk 2]
```diff
@@ -1,33 +1,36 @@
...
   <% end
-     if @node.fetch('haproxy',{}).fetch('ssl',{}) == 'only' -%>
+     if haproxy_cfg.fetch('ssl',{}) == 'only' -%>
```
Comment (till) [inline haproxy/templates/default/haproxy.cfg.frontend.erb]: @fh Sort of the same here. Default is an object. Maybe it should default to whatever "ssl" could be? Or what the default is for that.

File: haproxy/templates/default/haproxy.cfg.frontend.erb (hunk 1 above)
Comment (fh) [inline haproxy/templates/default/haproxy.cfg.frontend.erb]: Yes, agreed to both

File: haproxy/templates/default/haproxy.cfg.frontend.erb (hunk 2 above)
Comment (fh) [inline haproxy/templates/default/haproxy.cfg.frontend.erb]: The default is "off", I guess, yes.

File: haproxy/templates/default/haproxy.cfg.frontend.erb [hunk 3]
```diff
@@ -1,33 +1,36 @@
...
     <% end -%>
   <% end -%>
 
-  <% if !@node.fetch('haproxy',{}).fetch('additional_layers',{}).empty?-%>
+  <% if !haproxy_cfg.fetch('additional_layers',{}).empty?-%>
   #additional backends:
-    <% @node['haproxy']['additional_layers'].each do |layername,layerconfig| -%>
-      <% if @node.fetch('opsworks', {}).fetch('layers', {}).fetch(layername, {}).fetch('instances', {}).nil? -%>
+    <% haproxy_cfg['additional_layers'].each do |layername,layerconfig| -%>
+      <% if opsworks_layers.fetch(layername, {}).fetch('instances', {}).nil? -%>
```
Comment (fh) [inline haproxy/templates/default/haproxy.cfg.frontend.erb]: This also should be a check for empty, not nil

**Comment by till** [comment]
@fh addressed the comments.

**Pull request merged**